### PR TITLE
[Refactor] From XContentType.JSON to MediaTypeRegistry.JSON

### PR DESCRIPTION
**Pull request by nknize**
This PR rote refactors nearly all instances and usages of the `XContentType.JSON` singleton instance defined in the `:libs:opensearch-x-content` library to the `MediaTypeRegistry.JSON` singleton instance defined in `:libs:opensearch-core`. This decouples the server and test implementations from the x-content library such that foundation classes can be further refactored from the `:server` module to the `:opensearch-core` library without requiring a strict dependency on the x-content library. This keeps opensearch-core library loosely coupled with the remaining `:libs:opensearch-*` libraries such that serverless and cloud-native implementations can selective import only the libraries needed for implementation.

relates #5910
relates #8110

### CODE REVIEW

**Comment by nknize** [comment]
Flakey shard allocation failure tests.

File: client/rest-high-level/src/main/java/org/opensearch/client/RequestConverters.java [hunk 1]
```diff
@@ -266,7 +267,12 @@ static Request bulk(BulkRequest bulkRequest) throws IOException {
                     }
                 }
             } else if (opType == DocWriteRequest.OpType.UPDATE) {
-                source = XContentHelper.toXContent((UpdateRequest) action, bulkContentType, ToXContent.EMPTY_PARAMS, false).toBytesRef();
+                source = org.opensearch.core.xcontent.XContentHelper.toXContent(
```
Comment (reta) [inline client/rest-high-level/src/main/java/org/opensearch/client/RequestConverters.java:270]:
```suggestion
                import org.opensearch.core.xcontent.XContentHelper;
```
?

Seems like we could eliminate unnecessary formatting changes (noise on the pull request)

Comment (nknize) [reply client/rest-high-level/src/main/java/org/opensearch/client/RequestConverters.java:270]: Like the refactoring of the large `Strings.*`, this is incrementally refactoring `XContentHelper` so a little further below you'll see the following which is in the `org.opensearch.common.xcontent.XContentHelper` and not in the `org.opensearch.core.xcontent.XContentHelper`

```
                    XContentParser parser = XContentHelper.createParser(
                        /*
                         * EMPTY and THROW are fine here because we just call
                         * copyCurrentStructure which doesn't touch the
                         * registry or deprecation.
                         */
                        NamedXContentRegistry.EMPTY,
                        DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
                        indexSource,
                        mediaType
                    )
```

File: server/src/main/java/org/opensearch/action/bulk/BulkProcessor.java [hunk 2]
```diff
@@ -457,12 +458,8 @@ public BulkProcessor add(BytesReference data, @Nullable String defaultIndex, XCo
     /**
      * Adds the data from the bytes to be processed by the bulk processor
      */
-    public BulkProcessor add(
-        BytesReference data,
-        @Nullable String defaultIndex,
-        @Nullable String defaultPipeline,
-        XContentType xContentType
-    ) throws Exception {
+    public BulkProcessor add(BytesReference data, @Nullable String defaultIndex, @Nullable String defaultPipeline, MediaType xContentType)
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/bulk/BulkProcessor.java:461]:
```suggestion
    public BulkProcessor add(BytesReference data, @Nullable String defaultIndex, @Nullable String defaultPipeline, MediaType mediaType)
```

Comment (nknize) [reply server/src/main/java/org/opensearch/action/bulk/BulkProcessor.java]: ✅

**Comment by reta** [comment]
Test failures look legit

**Comment by nknize** [comment]
> Test failures look legit

They were.. it was a bonehead typo on my part. :/
